### PR TITLE
Handle self user registration failure when the email domain is not matched with organization email domain

### PR DESCRIPTION
**Pull request by sadilchamishka**
### Proposed changes in this pull request

1. Enable self registration for orgs

```
[identity_mgt.user_self_registration]
allow_self_registration=true
lock_on_creation=true
enable_recaptcha=true
verification_email_validity="1440"
callback_url="[${carbon.protocol}://${carbon.host}:${carbon.management.port}].*[/authenticationendpoint/login.do]*"
enable_resend_confirmation_recaptcha=true

[identity_mgt.user_self_registration.notification]
manage_internally=true   
```

2. Set email domain organization discovery for an organization.

3. Try self registering with setting email as not matching email domain

### Related Issues
- https://github.com/wso2/product-is/issues/19911

### After the fix

Screenshot 2024-10-02 at 16 45 20

### CODE REVIEW

**Comment by sadilchamishka** [comment]
Better if we can have validation logic at the portal level. But raising the PR as the previous server error and the display msg looks not intuitive for the end user.

**Review by PasinduYeshan — APPROVED** [review]

**Review by AnuradhaSK — APPROVED** [review]

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/IdentityRecoveryConstants.java [hunk 1]
```diff
@@ -338,6 +338,7 @@ public enum ErrorMessages {
                 "requirements of the selected registration option."),
         ERROR_CODE_INVALID_REGISTRATION_OPTION("USR-10007", "Invalid registration option."),
         ERROR_CODE_MULTIPLE_REGISTRATION_OPTIONS("USR-10008", "Multiple registration options are not supported."),
+        ERROR_CODE_INVALID_DOMAIN("USR-10009", "The email domain is not match with organization email domain."),
```
Comment (PasinduYeshan) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/IdentityRecoveryConstants.java:341]: Shall we refine the wording bit?

Comment (PasinduYeshan) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/IdentityRecoveryConstants.java:341]: The email domain does not match the organization's email domain.

Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/IdentityRecoveryConstants.java]: Fixed by - https://github.com/wso2-extensions/identity-governance/pull/813/commits/da9ce97ff9cdc7b0df161dc63072353d943edfc1

**Review by PasinduYeshan — APPROVED** [review]

**Pull request merged**